templatestore/frontend/src/pages/templateScreen.js

```diff
@@ -153,41 +161,87 @@ class TemplateScreen extends Component {
                         config: response.data
                     });
                 }
+                this.updateurlKeyList();
             })
             .catch(function(error) {
                 console.log(error);
             });
+        this.settinyUrlObj();
+    }
+
+    scan(parent, obj, result) {
+        var k;
+        if (obj instanceof Object) {
+            for (k in obj) {
+                if (obj.hasOwnProperty(k)) {
+                    //recursive call to scan property
+                    parent.push(k);
+                    this.scan(parent, obj[k], result);
+                    parent.pop();
+                }
+            }
+        } else if (validateURL(obj)) {
+            let name = generateNameOfUrl(parent);
+            result.push(name);
+        }
+        return;
+    }
+
+    settinyUrlObj() {
+        let data = {
+            name: this.state.templateData.name,
+            version: this.state.templateData.version
+        };
+        axios({
+            method: 'post',
```

Comment:
**get_tiny_url_from_db** -> change this.

Reply:
Any alternatives?

Reply:
/api/tiny_url